vendor/engines/saml_authentication/lib/saml_authentication/devise_configurator.rb

```diff
@@ -0,0 +1,59 @@
+require "saml_authentication/user_locator"
+require "saml_authentication/user_updater"
+
+module SamlAuthentication
+
+  class DeviseConfigurator
+
+    def configure!
+      Devise.setup do |config|
+        config.saml_session_index_key = :session_index
+        config.saml_default_user_key = :username
+        config.saml_create_user = true
+        config.saml_update_user = true
+        config.saml_resource_locator = SamlAuthentication::UserLocator.new
+        config.saml_update_resource_hook = SamlAuthentication::UserUpdater.new
+
+        config.saml_config = fetch_metadata_config
+
+        config.saml_configure do |settings|
+          settings.assertion_consumer_service_url = Rails.application.routes.url_helpers.auth_saml_user_session_url
+          settings.issuer = Rails.application.routes.url_helpers.metadata_saml_user_session_url
+
+          configure_security(settings)
+        end
+      end
+    end
+
+    private
+
+    def fetch_metadata_config
+      idp_metadata_parser = OneLogin::RubySaml::IdpMetadataParser.new
```

Comment:
This is going to need to change for Dartmouth, right? Can we point this to a centralized place where all the school-specific settings will live?

Reply:
This won't need to change. This class parses the URL configured in `Settings.saml.idp_metadata` which does all the school-specific setup.